api/src/main/java/org/apache/iceberg/DeleteFiles.java

```diff
@@ -81,4 +81,8 @@ default DeleteFiles deleteFile(DataFile file) {
    * @return this for method chaining
    */
   DeleteFiles caseSensitive(boolean caseSensitive);
+
+  DeleteFiles validateFilesExist(boolean validateFilesToDeleteExist);
+
+  DeleteFiles validateFromSnapshot(long snapshot);
```

Comment:
The implementation currently leverages MergingSnapshotProducer#validateDataFilesExist to determine that the files exist and weren't removed concurrently. That requires specifying a `startingSnapshotId` as the starting point for determining the relevant history to look at when doing the conflict detection logic (in this case checking if there are any new manifests since startingSnapshotId which have entries indicating that the file was already removed).

Callers of the DeleteFiles API are not required to provide the `startingSnapshotId`. By default in the implementation the `validateFromSnapshot` is null but this means that during the conflict validation check, it'll go through a longer history and read manifests which maybe would not need to be read anyways. So exposing this option gives callers a chance to check a narrower range if they know a good starting point to validate from.

Reply:
Oh I maybe see what you're getting at, do you mean that it's sufficient to simply just check the file existence (just a fileIO existence check) instead of leveraging the validateDataFilesExist method? Sounds reasonable. I was originally thinking we'd want to go through the manifests, because those would be legitimate operations on Iceberg tables. As opposed to someone not going through Iceberg and removing files and corrupting their table through some other process.

In the end file existence is the only thing we care about in terms of validation for the delete to succeed

Reply:
I think there's a bit of a misunderstanding. I thought that this feature would call `failMissingDeletePaths()`, which uses the delete managers and avoids the need to check metadata changes since some point in the past.

```java
  protected void failMissingDeletePaths() {
    filterManager.failMissingDeletePaths();
    deleteFilterManager.failMissingDeletePaths();
  }
```

Reply:
I stepped through how `failMissingDeletePaths` works, yeah that is sufficient for the validation we want to perform here. That already validates the files that we want to delete still exist based on the manifests in the snapshot we are trying to commit on top of. That's all that's required